mlir/include/mlir/Dialect/LLVMIR/LLVMAttrDefs.td

```diff
@@ -47,6 +47,15 @@ def LinkageAttr : LLVM_Attr<"Linkage", "linkage"> {
   let assemblyFormat = "`<` $linkage `>`";
 }
 
+//===----------------------------------------------------------------------===//
+// FramePointerKindAttr
+//===----------------------------------------------------------------------===//
+
+def FramePointerKindAttr : LLVM_Attr<"FramePointerKind", ""> {
+  let parameters = (ins "framePointerKind::FramePointerKind":$framePointerKind);
+  let assemblyFormat = "$framePointerKind";
```

Comment:
```suggestion
  let assemblyFormat = "`<` $framePointerKind `>`";
```

Apparently, `<` `>` are required around the assembly format to make this attribute "simple enough for the pretty form".  https://github.com/llvm/llvm-project/blob/59e24eea3438ab2925bda025f52d2cce92195065/mlir/lib/IR/AsmPrinter.cpp#L2090-L2105

Reply:
I do not think this is the issue, since I have tried this and it only adds angle brackets.
`llvm.func @frame_pointer_func() vscale_range(1, 2) attributes {frame_pointer = #llvm<<"non-leaf">>} {`

Reply:
You didn't provide the mnemonic?

What about `def FramePointerKindAttr : LLVM_Attr<"FramePointerKind", "fp"> {`?

Reply:
Thanks for the suggestion, @joker-eph!

As mentioned above, even if the mnemonic is set to either 'fp' or 'framePointerKind', the results is still `#llvm.[mnemonic]<[value]> `, repectively (#llvm.fp<"non-leaf">, #llvm.framePointerKind<"non-leaf">). 

Setting the mnemonic to `framePointerKind` and adding `'<' '>'` makes lowering from MLIR to LLVM IR to work appropriately but in MLIR the parameter has to be assigned as `llvm.func @frame_pointer_func() attributes {frame_pointer = #llvm.framePointerKind<"non-leaf">} `. 

is this the right behaviour? If so, I will go ahead and update the patch.

Reply:
This matches what we want, I think.

Reply:
I would rather not mix inherent attribute with discardable attributes. But that's not gonna be automatic: you need to modify LLVMFuncOp::print() / LLVMFuncOp::parse() implementations.

Reply:
@joker-eph this is what was originally implemented.
@gysit and I previously suggested to not do that, to avoid bloating the printed form further. There are already inherent attributes that are mixed with the discardable attributes.
I suggest to ship this PR with the current approach and decide if we should add special parsing and printing for all inherent attributes or not after that.

Reply:
I don’t quite understand how pulling this before the { would be more bloated than after?

Reply:
The advantage of printing things in the attribute dictionary is that all attributes print the same way: "attribute_name = attribute_value". If it is printed outside, the function to print and parse is implemented by hand and it seems a bit harder to ensure consistency / problems with ordering during parsing etc.

If we want to clearly separate, discardable and inherent attributes we may also print it outside of the dictionary. I that case I would suggest to follow the style of vscale/comdat etc, i.e., `frame_pointer(non-leaf)` or similar

Reply:
In the end, should we proceed with the latest implementation or the first submission? I understand that there are certain advantages and disadvantages but which one would be the one desired, just to confirm.

Reply:
I would suggest to proceed with the latest implementation, but let's wait to see if @joker-eph agrees.